examples/ds-example.cpp

```diff
@@ -39,19 +39,19 @@ int main(int argc, char* argv[])
     deepstream::PocoWSHandler wsh;
     deepstream::Client client(uri, wsh, errh);
 
-    client.login("{}", [](const std::unique_ptr<deepstream::Buffer> &){
+    client.login(Buffer("{}"), [](const std::unique_ptr<deepstream::Buffer> &&){
             std::cout << "Client logged in" << std::endl;
             });
 
     // Subscribe to the event "adam"
-    Event::SubscribeFnPtr event_sub_ptr = client.event.subscribe(Buffer("adam"), [&](const Buffer& buff) {
+    const deepstream::SubscriptionId event_sub_id = client.event.subscribe(Buffer("adam"), [&](const Buffer& buff) {
```

Comment:
https://github.com/deepstreamIO/deepstream.io-client-cpp/blob/a38999727f5a7aa870e388219e48ecb8ff35f584/include/deepstream/core/client.hpp#L39
A typedef is just an alias/renaming of a type so `unsigned long` is valid anywhere that `SubscriptionId` is, albeit less descriptively, without requiring a type cast.